Partition blocked by endless RocksDB retry loop `Caught recoverable exception`

### <!-- Component -->

<!-- Zeebe- -->

### <!-- Affected version -->

<!-- -8.9 -->

### <!-- Severity -->

<!-- High- -->

### <!-- Likelihood -->

<!-- Unknown_ -->

### Description

Partition is unhealthy and [endless looping](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22camunda-benchmark%22%0Aresource.labels.location%3D%22europe-west1%22%0Aresource.labels.cluster_name%3D%22camunda-benchmark%22%0Aresource.labels.namespace_name%3D%22c8-medic-2026-16-45910311-realistic%22%0Alabels.k8s-pod%2Fapp%3D%22camunda-platform%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fcomponent%3D%22zeebe-broker%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Finstance%3D%22c8-medic-2026-16-45910311-realistic%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fmanaged-by%3D%22Helm%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fname%3D%22camunda-platform%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fpart-of%3D%22camunda-platform%22;pinnedLogId=2026-04-14T08:46:04.926469065Z%2F7uye5jod7qxvvxkg;cursorTimestamp=2026-04-14T08:46:04.926469065Z;duration=PT5M?project=camunda-benchmark) with:

```
io.camunda.zeebe.db.ZeebeDbException: Unexpected error occurred during RocksDB transaction commit.
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.commit(ZeebeTransaction.java:145)
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.lambda$updateState$17(ProcessingStateMachine.java:673)
	at io.camunda.zeebe.scheduler.retry.ActorRetryMechanism.run(ActorRetryMechanism.java:28)
	at io.camunda.zeebe.scheduler.retry.RecoverableRetryStrategy.run(RecoverableRetryStrategy.java:52)
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:84)
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:43)
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:125)
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:127)
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:104)
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:224)
Caused by: org.rocksdb.RocksDBException: Transaction could not check for conflicts for operation at SequenceNumber 426542457 as the MemTable only contains changes newer than SequenceNumber 426591839.  Increasing the value of the max_write_buffer_size_to_maintain option could reduce the frequency of this error.
	at org.rocksdb.Transaction.commit(Native Method)
	at org.rocksdb.Transaction.commit(Transaction.java:216)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.commitInternal(ZeebeTransaction.java:166)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.commit(ZeebeTransaction.java:141)
	... 9 more
```

<img width="1877" height="300" alt="Image" src="https://github.com/user-attachments/assets/fa942f46-f13c-44e9-97cb-2cc80c93637d" />
<img width="1882" height="861" alt="Image" src="https://github.com/user-attachments/assets/17cbbd50-bb10-400f-bc80-098175697a4b" />
<img width="1855" height="762" alt="Image" src="https://github.com/user-attachments/assets/f6006f00-b11e-464b-9d63-da1eb1296348" />

### Steps to reproduce

1. Run a load test
2. Not clear what caused this


### Current behavior

It is endlessly looping with this error, but it is not recovering.

### Expected behavior

Not sure - if it is recoverable i guess it should recover? If not we should fail.

### Workaround

1. do this
2. do that
3. now this happens and circumvents the bug


### Environment

SM

### Version

- Component Version: [e.g. 2.0.0]


### Rootcause

_No response_

### Solution ideas

_No response_

### Dev -> QA handover

- Resources:
- Versions to validate:
- Release version (in which version this feature will be released):


### Links

_No response_

### Automated Test Impact

_No response_

--- a/zeebe/scheduler/src/main/java/io/camunda/zeebe/scheduler/retry/RecoverableRetryStrategy.java
+++ b/zeebe/scheduler/src/main/java/io/camunda/zeebe/scheduler/retry/RecoverableRetryStrategy.java
@@ -12,6 +12,8 @@ import io.camunda.zeebe.scheduler.future.ActorFuture;
 import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
 import io.camunda.zeebe.scheduler.retry.ActorRetryMechanism.Control;
 import io.camunda.zeebe.util.exception.RecoverableException;
+import io.camunda.zeebe.util.logging.ThrottledLogger;
+import java.time.Duration;
 import java.util.function.BooleanSupplier;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -19,6 +21,8 @@ import org.slf4j.LoggerFactory;
 public final class RecoverableRetryStrategy implements RetryStrategy {
 
   private static final Logger LOG = LoggerFactory.getLogger(RecoverableRetryStrategy.class);
+  private static final ThrottledLogger THROTTLED_LOG =
+      new ThrottledLogger(LOG, Duration.ofSeconds(5));
 
   private final ActorControl actor;
   private final ActorRetryMechanism retryMechanism;
@@ -56,7 +60,7 @@ public final class RecoverableRetryStrategy implements RetryStrategy {
       }
     } catch (final RecoverableException ex) {
       if (!terminateCondition.getAsBoolean()) {
-        LOG.debug("Caught recoverable exception, will retry: {}", ex.getMessage(), ex);
+        THROTTLED_LOG.warn("Caught recoverable exception, will retry: {}", ex.getMessage(), ex);
         actor.run(this::run);
         actor.yieldThread();
       }

--- a/zeebe/scheduler/src/main/java/io/camunda/zeebe/scheduler/retry/RecoverableRetryStrategy.java
+++ b/zeebe/scheduler/src/main/java/io/camunda/zeebe/scheduler/retry/RecoverableRetryStrategy.java
@@ -21,11 +21,10 @@ import org.slf4j.LoggerFactory;
 public final class RecoverableRetryStrategy implements RetryStrategy {
 
   private static final Logger LOG = LoggerFactory.getLogger(RecoverableRetryStrategy.class);
-  private static final ThrottledLogger THROTTLED_LOG =
-      new ThrottledLogger(LOG, Duration.ofSeconds(5));
 
   private final ActorControl actor;
   private final ActorRetryMechanism retryMechanism;
+  private final ThrottledLogger throttledLog = new ThrottledLogger(LOG, Duration.ofSeconds(5));
   private CompletableActorFuture<Boolean> currentFuture;
   private BooleanSupplier terminateCondition;
 
@@ -60,7 +59,7 @@ public final class RecoverableRetryStrategy implements RetryStrategy {
       }
     } catch (final RecoverableException ex) {
       if (!terminateCondition.getAsBoolean()) {
-        THROTTLED_LOG.warn("Caught recoverable exception, will retry: {}", ex.getMessage(), ex);
+        throttledLog.warn("Caught recoverable exception, will retry: {}", ex.getMessage(), ex);
         actor.run(this::run);
         actor.yieldThread();
       }

